python/plugins/processing/core/parameters.py

```diff
@@ -434,14 +441,19 @@ def getAsScriptCode(self):
 class ParameterNumber(Parameter):
 
     def __init__(self, name='', description='', minValue=None, maxValue=None,
-                 default=0.0, optional=False):
-        Parameter.__init__(self, name, description, optional)
-        try:
-            self.default = int(unicode(default))
-            self.isInteger = True
-        except:
-            self.default = float(default)
+                 default=None, optional=False):
```

Comment:
Except for Boolean and Selection, all default are set to None because I think code don't have to force default value.